src/mlpack/tests/main_tests/det_test.cpp

```diff
@@ -0,0 +1,381 @@
+
+  SetInputParam("input_model", CLI::GetParam<DTree<>*>("output_model"));
+  SetInputParam("test", std::move(testData));
+
+  mlpackMain();
+
+  // Check that initial estimates and final estimate using saved model are same.
+  CheckMatrices(training_set_estimates,
+                CLI::GetParam<arma::mat>("training_set_estimates"));
+  CheckMatrices(test_set_estimates,
+                CLI::GetParam<arma::mat>("test_set_estimates"));
+}
+
+/**
+ * Check that number of output variable importance values equals number of
+ * features in input data.
+ */
+BOOST_AUTO_TEST_CASE(DETViDimensionTest)
+{
+  arma::mat trainingData;
+  if (!data::Load("iris.csv", trainingData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  arma::mat testData;
+  if (!data::Load("iris_test.csv", testData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  size_t testRows = testData.n_rows;
+
+  // Input data.
+  SetInputParam("training", std::move(trainingData));
+  SetInputParam("test", std::move(testData));
+
+  mlpackMain();
+
+  // Check the number of output points equals number of input features.
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("vi").n_rows, 1);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("vi").n_cols, testRows);
+}
+
+/**
+ * Make sure only one of training data or pre-trained model is passed.
+ */
+BOOST_AUTO_TEST_CASE(DETModelValidityTest)
+{
+  arma::mat trainingData;
+  if (!data::Load("iris.csv", trainingData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("training", std::move(trainingData));
+
+  mlpackMain();
+
+  SetInputParam("input_model", CLI::GetParam<DTree<>*>("output_model"));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check learning process using different min_leaf_size.
+ */
+BOOST_AUTO_TEST_CASE(DETDiffMinLeafTest)
+{
+  arma::mat trainingData;
+  if (!data::Load("iris.csv", trainingData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  arma::mat testData;
+  if (!data::Load("iris_test.csv", testData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  // Input data.
+  SetInputParam("training", trainingData);
+  SetInputParam("test", testData);
+
+  mlpackMain();
+
+  arma::mat training_set_estimates =
+    CLI::GetParam<arma::mat>("training_set_estimates");
+  arma::mat test_set_estimates = CLI::GetParam<arma::mat>("test_set_estimates");
+
+  bindings::tests::CleanMemory();
+
+  // Train model using min_leaf_size equals to 10.
+
+  SetInputParam("training", std::move(trainingData));
+  SetInputParam("test", std::move(testData));
+  SetInputParam("min_leaf_size", (int) 10);
+
+  mlpackMain();
+
+  // Check that initial estimates and final estimates using two models are
+  // different.
+  BOOST_REQUIRE_LT(arma::accu(training_set_estimates ==
+    CLI::GetParam<arma::mat>("training_set_estimates")),
+    training_set_estimates.n_elem);
```

Comment:
Another tiny style issue; the wrapped lines should be indented twice.  Sorry to have so many little picky comments...